### DISCUSSION TITLE
The usage of server generator

### DISCUSSION THREAD

**Discussion by zZHorizonZz** [quarkiverse/quarkus-openapi-generator#749] [Q&A]
Hi, I have a few questions about how to use the server stubs within an application. 

From my understanding, based on this issue [https://github.com/quarkiverse/quarkus-openapi-generator/issues/670#issuecomment-1991529447](https://github.com/quarkiverse/quarkus-openapi-generator/issues/670#issuecomment-1991529447), the stubs generated from OpenAPI specs should act as an intermediary layer (At least in Quarkus). This means you shouldn't use them directly at the endpoint definition level.  I came to this conclusion because it seems you can't return specific Response types (like a 201 Created status) directly from the stubs and also reactive stubs use `CompletionStage` which makes it bothersome to use in Quarkus because you need to convert `Uni` to `CompletionStage` most of the time.

However, I'm a bit confused why the intermediary layer would generate endpoint annotations like `@POST` and `@Path`.

My second question is about Mutiny support. Currently, when I set `quarkus.openapi.generator.reactive=true`, it generates methods with `CompletionStage` return types (Although for me it just doesn't seem to work for some reason). If I wanted to use Mutiny, I would need to convert between `Uni` and `CompletionStage`. For example gRPC generator generates code that returns `Uni` directly, which is more convenient.

Could someone clarify these points for me?

**Comment by hbelmiro** [comment]
cc @carlesarnal
